In [1]:
import pandas as pd
from scipy import stats
import numpy as np

**Question No. 01**

In [2]:
df= pd.read_csv('D:\\Data science\\Assignments\\A 3 Hypothesis Testing\\Cutlets.csv')

In [19]:
df.describe

<bound method NDFrame.describe of      Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]>

In this Question: 
Null Hypothersis(HO):- Unit A = Unit B,
Alternate Hypothesis(HA):- Unit A is not equal to Unit B

In [5]:
UnitA = df.iloc[:,0]
UnitA.head()

0    6.8090
1    6.4376
2    6.9157
3    7.3012
4    7.4488
Name: Unit A, dtype: float64

In [6]:
UnitB= df.iloc[:,1]
UnitB.head()

0    6.7703
1    7.5093
2    6.7300
3    6.7878
4    7.1522
Name: Unit B, dtype: float64

In [7]:
stats.ttest_ind(UnitA,UnitB)

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

pvalue is not significant to alpha

Here we can see that pvalue 0.4723 > 0.05 so We accept Null Hypothesis

**Hence UnitA = UnitB**

**Question No. 02**

In [8]:
df= pd.read_csv("D:\\Data science\\Assignments\\A 3 Hypothesis Testing\\LabTAT.csv")

In [20]:
df.describe

<bound method NDFrame.describe of      Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]>

In this Question: Null Hypothesis(HO):- All averages are Equal,

Alternate Hypothesis(HA):- All Averages are different

In [12]:
Lab1= df.iloc[:,0]
Lab2= df.iloc[:,1]
Lab3= df.iloc[:,2]
Lab4= df.iloc[:,3]

In [26]:
stats.f_oneway(Lab1,Lab2,Lab3,Lab4)

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

pvalue is significant to alpha 2.1167e-57 < 0.05 So we reject Null Hypothesis

**All Laboratory Averages are different**

**Question No.03**

In [27]:
df= pd.read_csv('D:\\Data science\\Assignments\\A 3 Hypothesis Testing\\Buyerratio.csv')

In [28]:
df.describe

<bound method NDFrame.describe of   Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750>

In this Question:

Null Hypothesis(HO):- All Properties are equal

Alternate Hypothesis(HA):- Not all Properties are equal

In [32]:
Sales= df.iloc[:,1:5]
Sales= Sales.values

In [33]:
Sales

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [35]:
stats.chi2_contingency(Sales)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

pvalue is not significant to alpha 0.660 > 0.05 So we accept Null Hypothesis

**All properties are equal**

**Question N0. 04**

In [36]:
df= pd.read_csv('D:\\Data science\Assignments\\A 3 Hypothesis Testing\\Costomer+OrderForm.csv')

In [38]:
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [37]:
df.describe

<bound method NDFrame.describe of     Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]>

In this question:

Null Hypothesis(HO):- Defective% varies by Centre

Alternate Hypothesis((HA):- Defective% does not Varies by Centre

In [45]:
df.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [46]:
df.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [47]:
df.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [48]:
df.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [49]:
Telecall = np.array([[271,267,269,280],[29,33,31,20]])

In [51]:
stats.chi2_contingency(Telecall)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

pvalue is not significant to alpha 0.2771 > 0.05 So we accept Null Hypothesis

**Defective % varies by Centre**